collector/profile_status.go

```diff
@@ -0,0 +1,44 @@
+package collector
+
+import (
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	mgo "gopkg.in/mgo.v2"
+	"gopkg.in/mgo.v2/bson"
+        "time"
+)
+
+var (
+	profile_count = prometheus.NewGaugeVec(prometheus.GaugeOpts{
+		Namespace: Namespace,
+		Subsystem: "profile",
+		Name:      "slow_query_30s_count",
+		Help:      "The number of slow queries in this database during last 30 seconds",
+	}, []string{"database"})
+)
+
+type ProfileStatus struct {
+	Name        string `bson:"database"`
+	Count       int  `bson:"count"`
+}
+
+func (profileStatus *ProfileStatus) Export(ch chan<- prometheus.Metric) {
+	profile_count.WithLabelValues(profileStatus.Name).Set(float64(profileStatus.Count))
+	profile_count.Collect(ch)
+	profile_count.Reset()
+}
+
+func (profileStatus *ProfileStatus) Describe(ch chan<- *prometheus.Desc) {
+	profile_count.Describe(ch)
+}
+
+func CollectProfileStatus(session *mgo.Session, db string, ch chan<- prometheus.Metric) {
+	ts := time.Now().Add(-time.Duration(time.Second * 24))
```

Comment:
why 24?

Reply:
Because I was testing it works, and tried 24 hours, sorry for that ;)